internal/pkg/scorecard/interfaces.go

```diff
@@ -21,64 +21,71 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
 	scplugins "github.com/operator-framework/operator-sdk/internal/pkg/scorecard/plugins"
 	scapiv1alpha1 "github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha1"
-
-	"github.com/spf13/viper"
+	"github.com/operator-framework/operator-sdk/version"
+	v1 "k8s.io/api/core/v1"
 )
 
 type Plugin interface {
 	Name() string
 	Run() scapiv1alpha1.ScorecardOutput
 }
 
-type genericPlugin struct {
-	binPath string
+type externalPlugin struct {
+	name   string
+	config externalPluginConfig
 }
 
-type internalPlugin struct {
+type basicOrOLMPlugin struct {
```

Comment:
Yeah, we discussed separating the `Run` functions in another PR. They are very similar and would only differ by about 5-10 lines out of a very large function. They also have all the same configuration fields (and they both use all of them except for `csv-path` in the basic plugins). Splitting up the `Run` function into multiple smaller functions would be good to make it clearer as to what's actually going on in the function and is something we definitely should do, but we still would essentially be running the same functions for both plugins.

Reply:
This isn't to say that no future internal plugins would be separate. For the proposed YAML-defined scorecard tests, we would have a completely separate `Run`: https://github.com/operator-framework/operator-sdk/pull/1241

Reply:
Oh right. Sorry I'm harping on it 🙂 

> They also have all the same configuration fields (and they both use all of them except for csv-path in the basic plugins)

For this we could use struct embedding where the common fields go in a `basicPlugin` struct and an `olmPlugin` struct embeds basicPlugin and adds `csv-path`.

> Splitting up the Run function into multiple smaller functions would be good to make it clearer as to what's actually going on in the function and is something we definitely should do

Yeah that's pretty much what I'm suggesting.

> but we still would essentially be running the same functions for both plugins.

Which is totally fine. I'm basically saying that code that runs based on a `pluginType` conditional could be extracted into the plugin's run function, and all other code could be shared in smaller split-out functions.

And again, probably not in scope for this PR, but something like this would be good to do in a follow-up depending on direction from CVP requirements.